### PR TITLE
[5.3] Added tests when response is not key-value pair array.

### PR DESCRIPTION
**Pull request by srmklive**
Added tests for JSON response where response was not a key-value pair format like:
```
['foo' => 'bar']
```

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php [hunk 1]
```diff
@@ -328,7 +328,7 @@ public function seeJson(array $data = null, $negate = false)
     }
 
     /**
-     * Assert that the response doesn't contain JSON.
+     * Assert that the response does not contains JSON.
```
Comment (ntzm) [inline src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php]: Why was this changed? It doesn't even make sense now

File: tests/Foundation/FoundationMakesHttpRequestsJsonTest.php [hunk 2]
```diff
@@ -81,6 +81,20 @@ public function testSeeJsonEqualsWithNonJson()
...
+
+    public function testSeeJsonResponseOnString()
+    {
+        $this->response = new Illuminate\Http\Response(['foo']);
+
+        try {
+            $this->dontSeeJson(['foo']);
+            $this->seeJsonContains(['foo']);
+        } catch (PHPUnit_Framework_AssertionFailedError $e) {
+            $this->assertStringStartsWith('Invalid JSON', $e->getMessage());
```
Comment (itsash--) [inline tests/Foundation/FoundationMakesHttpRequestsJsonTest.php]: Do we necessarily want to be testing exception messages?

I thought this was generally bad practice/frowned upon.

Comment (GrahamCampbell) [inline tests/Foundation/FoundationMakesHttpRequestsJsonTest.php]: We'd quite like to know if the exception message changes by mistake, so this is fine imo.

**Comment by taylorotwell** [comment]
I don't understand the change to this message. It makes less sense now than it did.
